apps/api/src/billing/services/usage/usage.service.spec.ts

```diff
@@ -0,0 +1,243 @@
+        ];
+
+        usageRepository.getHistory.mockResolvedValue(mockUsageData);
+        leaseRepository.countByOwner.mockRejectedValue(new Error("Lease DB error"));
+
+        await expect(service.getHistoryStats(address, startDate, endDate)).rejects.toThrow("Lease DB error");
+      });
+
+      it("should handle large numbers and rounding correctly", async () => {
+        const { address, startDate, endDate, service, usageRepository, leaseRepository } = setup();
+
+        const mockUsageData = [
+          BillingUsageSeeder.create({
+            date: "2024-01-01",
+            activeLeases: 1,
+            dailyAktSpent: 100.123456789,
+            totalAktSpent: 100.123456789,
+            dailyUsdcSpent: 50.987654321,
+            totalUsdcSpent: 50.987654321,
+            dailyUsdSpent: 151.11111111,
+            totalUsdSpent: 151.11111111
+          }),
+          BillingUsageSeeder.create({
+            date: "2024-01-02",
+            activeLeases: 1,
+            dailyAktSpent: 200.987654321,
+            totalAktSpent: 301.11111111,
+            dailyUsdcSpent: 75.123456789,
+            totalUsdcSpent: 126.11111111,
+            dailyUsdSpent: 276.11111111,
+            totalUsdSpent: 427.222222221
+          })
+        ];
+
+        usageRepository.getHistory.mockResolvedValue(mockUsageData);
+        leaseRepository.countByOwner.mockResolvedValue(1000);
+
+        const result = await service.getHistoryStats(address, startDate, endDate);
+
+        expect(result.totalSpent).toBe(427.22);
+        expect(result.averagePerDay).toBe(213.61);
+        expect(result.totalLeases).toBe(1000);
+        expect(result.averageLeasesPerDay).toBe(500.0);
+      });
+    });
+  });
+
+  function setup(usageData?: BillingUsageRawResult[], totalLeases?: number) {
+    const address = AkashAddressSeeder.create();
+    const startDate = faker.date.past();
+    const leasesCount = totalLeases ?? faker.number.int({ min: 1, max: 20 });
+
+    const mockUsageData = usageData || [
+      BillingUsageSeeder.create({
+        date: format(startDate, "yyyy-MM-dd"),
+        activeLeases: 1,
+        dailyAktSpent: 5.5,
+        totalAktSpent: 5.5,
+        dailyUsdcSpent: 2.25,
+        totalUsdcSpent: 2.25,
+        dailyUsdSpent: 7.75,
+        totalUsdSpent: 7.75
+      }),
+      BillingUsageSeeder.create({
+        date: format(addDays(startDate, 1), "yyyy-MM-dd"),
+        activeLeases: 2,
+        dailyAktSpent: 3.2,
+        totalAktSpent: 8.7,
+        dailyUsdcSpent: 1.5,
+        totalUsdcSpent: 3.75,
+        dailyUsdSpent: 4.7,
+        totalUsdSpent: 12.45
+      }),
+      BillingUsageSeeder.create({
+        date: format(addDays(startDate, 2), "yyyy-MM-dd"),
+        activeLeases: 1,
+        dailyAktSpent: 2.1,
+        totalAktSpent: 10.8,
+        dailyUsdcSpent: 1.0,
+        totalUsdcSpent: 4.75,
+        dailyUsdSpent: 3.1,
+        totalUsdSpent: 15.55
+      })
+    ];
+
+    const usageRepository = stub<UsageRepository>({
```

Comment:
suggestion(non-blocking): I think `stub` is an early helper which was added before `jest-mock-extended`. As we now have jest-mock-extended, we can use its `mock` helper

Reply:
Done